### PR TITLE
ci: update s-z workflows to use labeled on pull_request trigger

### PR DESCRIPTION
**Pull request by grayside**
## Description

Scoped to Samples workflows with file names starting with s-z.

Implementation of https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3747, sibling of https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/3748.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [ ] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: .github/workflows/workflows.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: workflows
+name: cloudworkflows
```
Comment (iennae) [inline .github/workflows/workflows.yaml:15]: ooh,  good catch to make this more clear.

**Comment by grayside** [comment]
- scheduler: initial failure with timeout. manual retry fixed
- talent: test failure from sample bug, possibly a race condition? manual retry fixed
- speech: test failure. retry failed
- translate: failed. retry failed
- videointelligence: failed. retry failed
- vision: failed, retry failed
- vision-productSearch: failed, retry failed

File: .github/workflows/workflows.yaml (hunk 1 above)
Comment (grayside) [reply .github/workflows/workflows.yaml:15]: This was triggering GHA lint errors in vscode.

**Comment by grayside** [comment]
These test failures are out of scope. Assembled notes to file follow-up bugs.

**Review by iennae — APPROVED** [review]
Thank you! acknowledging that the test failures will be resolved later.

**Pull request merged**